src/tree.rs

```diff
@@ -90,12 +87,50 @@ where
     }
 
     /// Get the root of the merkle tree.
-    ///
-    /// It is none if the tree is empty.
     pub fn root(&self) -> Ref<T> {
         self.root.item(0)
     }
 
+    /// Returns the root of the smallest sub-tree that holds all the leaves.
+    ///
+    /// Returns None when the tree is empty. In this case use
+    /// `T::EMPTY_SUBTREES[H - 1]` for the root instead.
+    // Allowing for missing docs on panic, since panic is impossible. See
+    // comment below.
+    #[allow(clippy::missing_panics_doc)]
+    pub fn smallest_subtree(&self) -> (Option<Ref<T>>, usize) {
+        let mut smallest_node = &self.root;
+        let mut next_node = &self.root;
+        let mut height = H;
+        loop {
+            let mut children = 0;
+            for child in smallest_node.children.iter() {
+                if child.is_some() {
+                    children += 1;
+                    // we can safely unwrap since we checked that child is
+                    // Some()
+                    next_node = child.as_ref().unwrap().as_ref();
```

Comment:
You can avoid to unwrap if you flatten the iterator

Reply:
True, i pushed a version with `flatten`. I think it's indeed nicer